### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Samuellekpor**
In this feature branch, I : 

1. Create a class Classroom
2. Create the has-many/belongs-to relationship between Classroom and Student
3. Create a class Book
4. Create a class Rental
5. Create the many-to-many relationship between Person and Book

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (KDlamini) [inline book.rb]: In the Book class, kindly make sure that there's a method for adding rentals where the `book` itself is sent as a parameter to create the new rental achieving the 'has-many' association.

For example:

```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

Remember to pay attention to the order of arguments in your Rental class. The code above is just a suggestion, therefore, you must modify it to fit the flow of your application.

File: person.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   def correct_name
```
Comment (KDlamini) [inline person.rb]: Same here, kindly make sure that there's a method for adding rentals where the `person` itself is sent as a parameter to create the new rental achieving the 'has-many' association.

- The method is similar to the one in the example above except now `self` should be the argument for the `person` parameter in the Rental instance.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @Samuellekpor 🙋‍♀️,  

Great work so far! 💯 
You have done a great job setting up associations for the OOP school library. 🎊 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Correct PR title and description
✔️  `has-many` / `belongs-to` relationships between `Classroom` and `Student` are created
✔️  Commit messages are meaningful
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Samuellekpor, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**